Source/Details/ASPageTable.mm

```diff
@@ -0,0 +1,149 @@
+    return @[];
+  }
+  
+  ASPageCoordinates *minPage = [ASPageCoordinates pageCoordinatesForPageThatContainsPoint:CGPointMake(CGRectGetMinX(rect), CGRectGetMinY(rect)) pageSize:pageSize];
+  ASPageCoordinates *maxPage = [ASPageCoordinates pageCoordinatesForPageThatContainsPoint:CGPointMake(CGRectGetMaxX(rect), CGRectGetMaxY(rect)) pageSize:pageSize];
+  if (ASObjectIsEqual(minPage, maxPage)) {
+    return @[minPage];
+  }
+  
+  NSUInteger minX = minPage.x;
+  NSUInteger minY = minPage.y;
+  NSUInteger maxX = maxPage.x;
+  NSUInteger maxY = maxPage.y;
+  
+  NSMutableArray *results = [NSMutableArray array];
+  for (NSUInteger x = minX; x <= maxX; x++) {
+    for (NSUInteger y = minY; y <= maxY; y++) {
+      [results addObject:[ASPageCoordinates pageCoordinatesWithX:x Y:y]];
+    }
+  }
+  
+  return results;
+}
+
+- (CGRect)pageRectForPageSize:(CGSize)pageSize
+{
+  CGFloat pageWidth = pageSize.width;
+  CGFloat pageHeight = pageSize.height;
+  return CGRectMake(_x * pageWidth, _y * pageHeight, pageWidth, pageHeight);
+}
+
+- (BOOL)isEqualToPageCoordinates:(ASPageCoordinates *)coords
+{
+  return (coords != nil) && (_x == coords.x) && (_y == coords.y);
+}
+
+- (BOOL)isEqual:(id)other
+{
+  if (self == other) {
+    return YES;
+  }
+  if (! [other isKindOfClass:[ASPageCoordinates class]]) {
+    return NO;
+  }
+  return [self isEqualToPageCoordinates:other];
+}
+
+- (NSUInteger)hash
+{
+  return ASHash64ToNative(ASHashCombine(_x, _y));
+}
+
+@end
+
+@implementation NSMapTable (ASPageTableMethods)
+
++ (instancetype)pageTableWithValuePointerFunctions:(NSPointerFunctions *)valueFuncs
+{
+  static NSPointerFunctions *pageCoordinatesFuncs;
+  static dispatch_once_t onceToken;
+  dispatch_once(&onceToken, ^{
+    // Don't use NSPointerFunctionsObjectPointerPersonality here as 2 ASPageCoordinates objects with the same x and y coordinates should be equal.
+    pageCoordinatesFuncs = [NSPointerFunctions pointerFunctionsWithOptions:NSPointerFunctionsStrongMemory];
+  });
+  
+  return [[NSMapTable alloc] initWithKeyPointerFunctions:pageCoordinatesFuncs valuePointerFunctions:valueFuncs capacity:0];
+}
+
++ (ASPageTable *)pageTableForStrongObjectPointers
+{
+  static NSPointerFunctions *strongObjectPointerFuncs;
+  static dispatch_once_t onceToken;
+  dispatch_once(&onceToken, ^{
+    strongObjectPointerFuncs = [NSPointerFunctions pointerFunctionsWithOptions:NSPointerFunctionsStrongMemory | NSPointerFunctionsObjectPointerPersonality];
+  });
+  return [self pageTableWithValuePointerFunctions:strongObjectPointerFuncs];
+}
+
++ (ASPageTable *)pageTableForWeakObjectPointers
+{
+  static NSPointerFunctions *weakObjectPointerFuncs;
+  static dispatch_once_t onceToken;
+  dispatch_once(&onceToken, ^{
+    weakObjectPointerFuncs = [NSPointerFunctions pointerFunctionsWithOptions:NSPointerFunctionsWeakMemory | NSPointerFunctionsObjectPointerPersonality];
+  });
+  return [self pageTableWithValuePointerFunctions:weakObjectPointerFuncs];
```

Comment:
Note: Personality doesn't matter for values, since they're never hashed/equaled.